dll/win32/wtsapi32/wtsapi32.c

```diff
@@ -285,11 +285,30 @@ BOOL WINAPI WTSQuerySessionInformationA(
     LPSTR* Buffer,
     DWORD* BytesReturned)
 {
-    /* FIXME: Forward request to winsta.dll::WinStationQueryInformationA */
-    FIXME("Stub %p 0x%08x %d %p %p\n", hServer, SessionId, WTSInfoClass,
-        Buffer, BytesReturned);
+    if (!BytesReturned || !Buffer)
+    {
+        SetLastError(ERROR_INVALID_USER_BUFFER);
+        return FALSE;
+    }
 
-    return FALSE;
+    if (!WTSQuerySessionInformationW(hServer, SessionId, WTSInfoClass, (LPWSTR*) Buffer, BytesReturned))
+        return FALSE;
+
+    /* FIXME: reencode strings with ANSI codepage */
+    switch (WTSInfoClass)
+    {
+        case WTSConnectState:
+        case WTSClientProtocolType:
+            break;
+        default:
+        {
+            FIXME("Stub (ANSI) %p 0x%08x %d %p %p\n", hServer, SessionId, WTSInfoClass,
```

Comment:
A `FIXME` in Wine will print the function name, so `(ANSI)` is redundant. In ROS it will show the line number, so it can be inferred, but you'd either want it to be Wine-style or to print the full function name.